SchemaMigration/COSchemaMigrationDriver.m

```diff
@@ -42,28 +42,81 @@ static inline void addObjectForKey(NSMutableDictionary *dict, id object, NSStrin
 	[value addObject: object];
 }
 
+/**
+ * Returns some info on the metamodel state that the item entity/package/version
+ * comes from, namely the package/version pairs for the entity and all of its
+ * superclasses.
+ */
+- (NSDictionary *) versionsByDomainForItem: (COItem *)item
+{
+	NSDictionary *versionsByDomainByEntityTuple = [COSchemaMigration versionsByDomainByEntityTuple];
+	NSArray *key = @[item.packageName, @(item.entityVersion), item.entityName];
+	return versionsByDomainByEntityTuple[key];
+}
+
+
+
 - (NSSet *)domainsToMigrateForItem: (COItem *)item
 {
-	NSArray *domains = [item valueForAttribute: kCOObjectDomainsProperty];
 	ETEntityDescription *entity = [_modelDescriptionRepository descriptionForName: item.entityName];
 	BOOL isDeletedEntity = (entity == nil);
 	
 	/* For a deleted entity, the domain versions match between item and packages */
 	if (isDeletedEntity)
 	{
-		/* The first domain is the one owning the entity, the remaining domains
-		   own the parent entities. */
-		return S([domains firstObject]);
+		// FIXME: We hit this line when loading a CODictionary during the test suite,
```

Comment:
It's expected we hit this line for CODictionary. CODictionary doesn't have a corresponding ETEntityDescription. It is an entity name created at serialization time when encoutering a NSDictionary. At deserialization time, when an item is declared as CODictionary, it will be deserialized as a NSDictionary.